packages/edit-site/src/components/global-styles/shadows-edit-panel.js

```diff
@@ -301,18 +301,20 @@ function ShadowEditor( { shadow, onChange } ) {
 	const shadowParts = useMemo( () => getShadowParts( shadow ), [ shadow ] );
 
 	const onChangeShadowPart = ( index, part ) => {
-		shadowParts[ index ] = part;
-		onChange( shadowParts.join( ', ' ) );
+		const newShadowParts = shadowParts.map( ( p, i ) =>
+			i === index ? part : p
+		);
```

Comment:
Do we really need to call `.map()` over all items? I think an inline mutation should be cheaper, and allowed by the compiler:

```suggestion
		const newShadowParts = [ ...shadowParts ];
		newShadowParts[ index ] = part;
```